templates/setenv.sh.erb

```diff
@@ -1,123 +1,169 @@
+<%- if scope.call_function('versioncmp', [@version, '8.11.0']) > 0 -%>
+JVM_REQUIRED_ARGS='-Djava.awt.headless=true -Datlassian.standalone=JIRA -Dorg.apache.jasper.runtime.BodyContentImpl.LIMIT_BUFFER=true -Dmail.mime.decodeparameters=true -Dorg.dom4j.factory=com.atlassian.core.xml.InterningDocumentFactory'
+<% else -%>
+JVM_REQUIRED_ARGS='-Djava.awt.headless=true -Datlassian.standalone=JIRA -Dorg.apache.jasper.runtime.BodyContentImpl.LIMIT_BUFFER=true -Dmail.mime.decodeparameters=true'
+<%- end -%>
+
+
 
 # Uncomment this setting if you want to import data without notifications
 #
 <% if @disable_notifications -%>
-DISABLE_NOTIFICATIONS=" -Datlassian.mail.senddisabled=true -Datlassian.mail.fetchdisabled=true -Datlassian.mail.popdisabled=true"
+DISABLE_NOTIFICATIONS='-Datlassian.mail.senddisabled=true -Datlassian.mail.fetchdisabled=true -Datlassian.mail.popdisabled=true'
 <% else %>
-#DISABLE_NOTIFICATIONS=" -Datlassian.mail.senddisabled=true -Datlassian.mail.fetchdisabled=true -Datlassian.mail.popdisabled=true"
+#DISABLE_NOTIFICATIONS='-Datlassian.mail.senddisabled=true -Datlassian.mail.fetchdisabled=true -Datlassian.mail.popdisabled=true'
 <% end -%>
 
+
 #-----------------------------------------------------------------------------------
 #
 # In general don't make changes below here
 #
 #-----------------------------------------------------------------------------------
 
 #-----------------------------------------------------------------------------------
-# This allows us to actually debug GC related issues by correlating timestamps
-# with other parts of the application logs.
+# Prevents the JVM from suppressing stack traces if a given type of exception
+# occurs frequently, which could make it harder for support to diagnose a problem.
 #-----------------------------------------------------------------------------------
-JVM_EXTRA_ARGS="-XX:+PrintGCDateStamps"
-
-PRGDIR=`dirname "$0"`
+<%- if @jvm_type == 'openjdk-11' -%>
+JVM_EXTRA_ARGS='-Xlog:gc*:verbose_gc.log:time,uptime:filecount=15,filesize=10M -XX:+ExplicitGCInvokesConcurrent -XX:-OmitStackTraceInFastThrow -Djava.locale.providers=COMPAT <%= @jvm_extra_args_additional %>'
+<%- elsif @jvm_type == 'oracle-jdk-1.8' -%>
+JVM_EXTRA_ARGS='-XX:+PrintGCDateStamps -XX:+ExplicitGCInvokesConcurrent -XX:-OmitStackTraceInFastThrow -Djava.locale.providers=COMPAT <%= @jvm_extra_args_additional %>'
+<%- elsif @jvm_type == 'custom' -%>
+JVM_EXTRA_ARGS='<%= @jvm_extra_args %>'
+<%- end -%>
+
+CURRENT_NOFILES_LIMIT=$( ulimit -Hn )
+ulimit -Sn "${CURRENT_NOFILES_LIMIT}"
+ulimit -n $(( CURRENT_NOFILES_LIMIT > MIN_NOFILES_LIMIT  ? CURRENT_NOFILES_LIMIT : MIN_NOFILES_LIMIT ))
+
+PRGDIR=$(dirname "$0")
 cat "${PRGDIR}"/jirabanner.txt
 
 JIRA_HOME_MINUSD=""
-if [ "$JIRA_HOME" != "" ]; then
-    echo $JIRA_HOME | grep -q " "
-    if [ $? -eq 0 ]; then
+if [[ "${JIRA_HOME}" != "" ]]; then
```

Comment:
This bashism breaks on Ubuntu. Currently we don't run the acceptance tests on it, but in https://github.com/voxpupuli/puppet-jira/pull/334 I found out that it breaks.